integration_tests/test_builtin_len.py

```diff
@@ -7,4 +9,20 @@ def test_len():
     assert len("abcd") == 4
     assert len("") == 0
 
+    l: list[i32]
+    l = [1, 2, 3, 4]
+    assert len(l) == 4
+    l2: list[f64]
+    l2 = [1.2, 3.4, 5.6, 7.8, 9.0]
+    assert len(l2) == 5
+
+    l3: list[i32] = []
+    l3.append(1)
+    l3.append(2)
+    l3.append(3)
```

Comment:
Let's do 20-30 appends in a loop and then check if the `len(xyz_list)` gives correct output.